src/Microsoft.VisualStudio.Composition/Diagnostic/DiagnosticInfoCollector.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.VisualStudio.Composition.Diagnostic
+{
+    using System;
+    using System.Collections.Generic;
+
+    internal class DiagnosticInfoCollector
+    {
+        private readonly List<string> messages = new List<string>();
+
+        internal void Collect(string message)
+        {
+            this.messages.Add(message);
+        }
+
+        internal string GetAndClearInformation()
+        {
+            var response = string.Join(Environment.NewLine, this.messages.ToArray());
```

Comment:
No reason to call `.ToArray()` on the List. `string.Join` has an overload that takes an `IEnumerable<T>`.